### PR TITLE
[UI/UX]Update ControlsView with CornerClipper

### PR DESCRIPTION
**Pull request by ipavlidakis**
### 🎯 Goal

Improve corner radius for views in order to allow for simpler layout.

### 🛠 Implementation

Offsetting a view by some pixels in any direction is causing issues with other views where available space can't be easily calculated. This PR is fixing this by providing corner radius without the need for offset.

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should receive manual QA
- [ ] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (Docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamVideoSwiftUI/Utils/ClipCorners.swift [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                .clipShape(RoundedCorners(radius: radius, corners: corners))
+
+            backgroundColorView
+
+            content
+                .layoutPriority(1)
+        }
+    }
+
+    ///
```
Comment (martinmitrevski) [inline Sources/StreamVideoSwiftUI/Utils/ClipCorners.swift:39]: Something forgotten here?

File: Sources/StreamVideoSwiftUI/Utils/ClipCorners.swift [hunk 2]
```diff
@@ -0,0 +1,101 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+
+/// `CornerClipper` is a ViewModifier that clips a SwiftUI view
+/// to have rounded corners on specified sides.
+struct CornerClipper: ViewModifier {
```
Comment (martinmitrevski) [inline Sources/StreamVideoSwiftUI/Utils/ClipCorners.swift:10]: not related to the PR: we have similar thing for the chat SDK, we might want to introduce view utils package at some point.

File: DemoApp/Sources/Views/Controls/DemoControls.swift [hunk 3]
```diff
@@ -28,41 +29,29 @@ struct AppControlsWithChat: View {
...
-            VStack {
-                colors.callControlsBackground
-                    .frame(height: 30)
-                    .cornerRadius(24)
-                Spacer()
-            }
-            .offset(y: -15)
+        .cornerRadius(
+            cornerRadius,
+            corners: [.topLeft, .topRight],
```
Comment (martinmitrevski) [inline DemoApp/Sources/Views/Controls/DemoControls.swift:51]: why only top left and right? Shouldn't the bottom be done as well?

File: Sources/StreamVideoSwiftUI/Utils/ClipCorners.swift (hunk 1 above)
Comment (ipavlidakis) [reply Sources/StreamVideoSwiftUI/Utils/ClipCorners.swift]: oops :)

**Review by martinmitrevski — APPROVED** [review]

**Pull request merged**